Add docker container for MapR

AS186170 opened the pull request:
mapr52-code checkin for docker  container

singhash committed:
mapr52-code checkin

mapr52-code checkin for docker  container

sanjay990 commented:
Rewrite the commit message. Something like `Add docker container for MapR`

sanjay990 commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+FROM teradatalabs/centos6-java8-oracle
```
License header missing. Add them to other files too

AS186170 replied:
@sanjay990  License header added for the files

sanjay990 commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+FROM teradatalabs/centos6-java8-oracle
+MAINTAINER Teradata Docker Team <docker@teradata.com>
+
+
```
Remove blank line

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+MAINTAINER Teradata Docker Team <docker@teradata.com>
+
+
+#Add repo for mapr
```
Please adopt a consistent comment style throughout this commit. Other Dockerfiles all have a '#' followed by a space, followed by a comment. The comment should start with a capital letter or be in all caps. The other Dockerfiles have mixed style in terms of small vs large cap.

petroav replied:
Also, please stylize mapr as MapR in all your comments.

AS186170 replied:
done

sanjay990 commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+RUN rpm --import http://package.mapr.com/releases/pub/maprgpg.key
+
+#add user and password
+RUN adduser mapr
```
Adding user and creating a home directory can be merged into one `-d`. Also looks like you can merge many of these commands. Try to combine them as far as possible

sanjay990 commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+
+#Add repo for mapr
+ADD files/maprtech.repo /etc/yum.repos.d/maprtech.repo
+RUN yum update -y
```
If possible, combine all the software installation with this step and do this first.

petroav replied:
I don't know if I agree. Keeping the packages close to where they are configured makes more sense to me. Maybe the line below about installing utility packages can be merged here but I think the MapR and Python installation blocks should remain where they are.

sanjay990 commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+
+# RUN SETUP script
+RUN /root/setup.sh
+
```
Remove all these extra lines from all the files.

sanjay990 commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,41 @@
+
+
+#run hdfs commands
+hadoop fs -mkdir /user/hive-user
```
Can you combine these commands 
hadoop fs -mkdir /user/hive-user /user/hdfs-user

petroav replied:
+1

Also group them in some semantic way so they're easier to read.

sanjay990 commented on teradatalabs/mapr52-hive/files/socks-proxy.sh:
```diff
@@ -0,0 +1 @@
+../../../commons/socks-proxy.sh
```
Add new line

AS186170 replied:
@sanjay990  new line  is not required as per artur , its same in other containers also

sanjay990 commented on teradatalabs/mapr52-hive/files/supervisord.d/bootstrap.conf:
```diff
@@ -0,0 +1 @@
+../../../../commons/supervisord.d/bootstrap.conf
```
Add new line

ArturGajowy replied:
Those are symlinks, so there's no control over it / doesn't matter

AS186170 replied:
@sanjay990  new line  is not required as per artur , its same in other containers also

sanjay990 commented on teradatalabs/mapr52-hive/files/supervisord.d/mysql-metastore.conf:
```diff
@@ -0,0 +1 @@
+../../../../commons/supervisord.d/mysql-metastore.conf
```
Add new line

AS186170 replied:
@sanjay990  new line  is not required as per artur , its same in other containers also

ArturGajowy replied:
The symlinks got messed up somehow. Please replace all the files with similar content with symlinks to the files pointed by the paths in those files.

sanjay990 commented on teradatalabs/mapr52-hive/files/supervisord.d/socks-proxy.conf:
```diff
@@ -0,0 +1 @@
+../../../../commons/supervisord.d/socks-proxy.conf
```
Add new line

AS186170 replied:
@sanjay990  new line  is not required as per artur , its same in other containers also

sanjay990 commented on teradatalabs/mapr52-hive/files/wardenTracker.sh:
```diff
@@ -0,0 +1,54 @@
+done
+
+
+
```
Remove extra lines

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+ADD files/maprtech.repo /etc/yum.repos.d/maprtech.repo
+RUN yum update -y
+
+#install utility softwares
```
Software is a non-countable noun so softwares is incorrect. Either say "Install utility software" or "Install utility packages"

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+RUN yum update -y
+
+#install utility softwares
+RUN yum install iputils vim openssh-server openssh-clients sudo -y
```
I think it is more idiomatic for Linux commands to have options before the positional arguments so I think you should move the `-y` option to the front on all instances of `yum install` in this Dockerfile and other files.

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+#install utility softwares
+RUN yum install iputils vim openssh-server openssh-clients sudo -y
+
+#configure sshh
```
sshh is incorrect. Also stylize to SSH.

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+#get maprgpg key
+RUN rpm --import http://package.mapr.com/releases/pub/maprgpg.key
+
+#add user and password
```
I think this comment doesn't add much extra context as the commands themselves are self-explanatory. I would remove it.

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+RUN mkdir /opt/mapr/zkdata
+RUN chmod 777 /opt/mapr/zkdata
+RUN mkdir -p /mapr
+RUN echo "/home/mapr/storagefile" > /root/disk.txt
```
This is a line where a comment is definitely needed. Someone who isn't familiar with the setup won't know why this is being done. I think the following comment would be appropriate: "The /root/disk.txt is read by MapR to determine which disks it can use and the /home/mapr/storagefile is a flat file that acts as a disk"

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+RUN rpm --import http://package.mapr.com/releases/pub/maprgpg.key
+
+#add user and password
+RUN adduser mapr
```
I would add that commands should be grouped by user. First do all commands for one user, then all commands for the next one. That way it's easier to tell is something is missing for a user. Also separate each block with a newline, right now the block of text is hard to parse.

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+#install mapr 
+RUN yum install mapr-fileserver mapr-nfs mapr-nodemanager  mapr-cldb mapr-zookeeper mapr-resourcemanager mapr-historyserver  mapr-webserver  mapr-gateway -y
+
+#necessary configuration for mapr
```
I would re-phrase this as "Configure Zookeeper's data directory" followed by another block with a comment saying "Configure MapR's disks.txt"

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+RUN mkdir -p /mapr
+RUN echo "/home/mapr/storagefile" > /root/disk.txt
+
+
```
Extra newline.

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,55 @@
+RUN echo "/home/mapr/storagefile" > /root/disk.txt
+
+
+#install python
```
This comment is incorrect, it looks like you copy pasted incorrectly, the comment should say "Install supervisord".

AS186170 replied:
done

petroav commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+
+
+# HDFS PORTS
+EXPOSE 1004 1006 8020 50010 50020 50070 50075 50470
```
Can you audit all these ports and determine if they're used by MapR? Here's the list http://maprdocs.mapr.com/home/ReferenceGuide/MapRPorts.html?hl=ports,used,mapr

petroav commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+RUN yum install -y mysql-server mysql-connector-java \
+    && yum -y clean all && rm -rf /tmp/* /var/tmp/* 
+
+#install lsof
```
Command is self explanatory, would remove the comment.

petroav commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+#install lsof
+RUN yum install -y lsof
+
+# RUN SETUP script
```
Same, comment doesn't add much.

petroav commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,42 @@
+# SOCKS PORT
+EXPOSE 1180
+
+#run startup script
```
Comment doesn't add extra information, I would remove it.

petroav commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,41 @@
+dd if=/dev/zero of=/home/mapr/storagefile bs=1G count=10
+/opt/mapr/server/disksetup -M -F /root/disk.txt
+
+#create proxy setup for hive users
```
I think "Create Hive proxy users" is more accurate

petroav commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,41 @@
+touch /opt/mapr/conf/proxy/hive-user
+touch /opt/mapr/conf/proxy/hdfs-user
+
+#start zookeeper
```
Comment is unnecessary.

petroav commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,41 @@
+#start zookeeper
+service mapr-zookeeper start
+
+#start warden 
```
Comment is unnecessary.

petroav commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,41 @@
+
+#start warden 
+service mapr-warden start
+
```
Extra newline.

petroav commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,41 @@
+/opt/mapr/server/configure.sh -R
+
+#wait for warden to start all the services
+
```
Extra newline.

petroav commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,41 @@
+
+sh /root/wardenTracker.sh
+
+
```
Extra newline.

petroav commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,41 @@
+hadoop fs -mkdir /user/hive/warehouse
+hadoop fs -chmod 777 /user/hive
+hadoop fs -chmod 777 /user/hive/warehouse
+
```
Two extra newlines.

petroav commented on teradatalabs/mapr52-hive/files/setup.sh:
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
You should rename this file to something more descriptive than "setup.sh". I suggest "configure_mysql.sh"

AS186170 replied:
@petroav this filename is similar to the other images code

petroav replied:
Hmm, yea but for example `setup.sh` for `cdh5-hive-master` does a bunch of things, so it's OK to name it something generic like `setup.sh` whereas here all we're doing is configuring MySQL so I think we should rename it. Don't feel strongly about it though. I can also see the argument for being consistent with other image files. Up to you.

ArturGajowy replied:
let's leave it named `setup.sh`

petroav commented on teradatalabs/mapr52-hive/files/setup.sh:
```diff
@@ -0,0 +1,9 @@
+
+mysql_install_db
+/usr/bin/mysqld_safe &
+sleep 20
```
We should not be using sleeps. Can you see if you can use some other command in a loop to ensure that mysqld has correctly started?

AS186170 replied:
@petroav sleeps are there in other setup code , i just copied from there to avoid any discrepancies

petroav commented on teradatalabs/mapr52-hive/files/setup.sh:
```diff
@@ -0,0 +1,9 @@
+sleep 20
+/usr/bin/mysqladmin -u root password 'root'
+killall mysqld
+sleep 10
```
Same here

AS186170 replied:
@petroav inline with the other container code

petroav commented on teradatalabs/mapr52-hive/files/setup.sh:
```diff
@@ -0,0 +1,9 @@
+/usr/bin/mysqld_safe &
+sleep 20
+/usr/bin/mysqladmin -u root password 'root'
+killall mysqld
```
Why are you killing mysqld?

AS186170 replied:
@petroav mysqld is being killed because it will be restarted by supervisord.d
all the other docker containers are also doing same.
This part of code is only for setup of mysql password after which it will be started again by supervisord.d

petroav commented on teradatalabs/mapr52-hive/files/supervisord.conf:
```diff
@@ -0,0 +1,22 @@
+[supervisord]
```
Can't we use the configuration in the common subfolder of the project? I'm not too familiar how this work so maybe this is a question for @ArturGajowy.

petroav commented on teradatalabs/mapr52-hive/files/supervisord.conf:
```diff
@@ -0,0 +1,22 @@
+
+[supervisorctl]
+serverurl = unix:///tmp/supervisor.sock
+prompt = cdh-pseudo-distributed
```
If we do end up keeping this config maybe we should update this to say something other than cdh-pseudo-distributed?

ArturGajowy commented:
Please split the commit in two parts, one for the base image and another one for the -hive one

ArturGajowy commented:
Please add a `capabilities.txt` file so that the tests are run for the -hive image. See any ohter -hive image.

ArturGajowy reviewed (commented):
just a skim and two comments for now, please ping me when those are addressed

petroav commented on teradatalabs/mapr52-hive/files/wardenTracker.sh:
```diff
@@ -0,0 +1,54 @@
+   maprcliReady=$(maprcli service list -node $hname | grep 'ERROR (10009)' | wc -l)
+   Services=0
+else
+   Services=$(maprcli service list -node $hname | grep JobHistoryServer |awk '{$1=$1};1' | tr ' ' '\n' | tail -1f)
```
You're grepping for JobHistoryServer but the comment says CLDB.

petroav commented on teradatalabs/mapr52-hive/files/wardenTracker.sh:
```diff
@@ -0,0 +1,54 @@
+done
+
+#wait for NodeManager to start
+
```
Extra newline here and in the other code blocks.

petroav commented on teradatalabs/mapr52-hive/files/wardenTracker.sh:
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
```
I don't know enough bash to say if this is idiomatic so I'll leave the review here to @ArturGajowy.

petroav reviewed (changes requested):
As Sanjay pointed out, please re-write your commit message. In general, if you are committing for any Presto related project follow this guideline: http://chris.beams.io/posts/git-commit/.

A lot of my comments relate to style, which is just as important as content. In future pay attention to the style around you in the project and try emulate it as much as possible.

ArturGajowy commented:
Guys, any updates on this? Please ping me when the (actionable) comments made so far are applied

cc @petroav @AS186170

AS186170 commented:
@ArturGajowy i have started working on modification for the review comments today.Will let you know once completed.

singhash committed:
Add docker container for MapR

Add docker container for MapR

singhash committed:
Add review changes for MapR docker container

Add review changes for MapR docker container

singhash committed:
Rebasing the branch to master

Rebasing the branch to master for splitting mapr commits into two

singhash committed:
Add docker container for MapR without hive

Add docker container code for MapR-Base

singhash committed:
Add docker container for MapR with hive

Add docker container code for MapR-Hive

AS186170 commented:
@sanjay990 @petroav @ArturGajowy  Review comments are now complete for the branch

petroav commented on teradatalabs/mapr52-base/README.md:
```diff
@@ -0,0 +1,9 @@
+# mapr52-base
+            
+
+Docker image with Hive installed from MapR repositories.
```
I don't think this is correct. This README is in the mapr52-base folder so this image shouldn't have Hive installed.

ArturGajowy replied:
+1. Just base this readme on https://github.com/Teradata/docker-images/blob/master/teradatalabs/iop-base/README.md

petroav commented on teradatalabs/mapr52-base/README.md:
```diff
@@ -0,0 +1,9 @@
+# mapr52-base
+            
+
```
Extra newline.

petroav commented on teradatalabs/mapr52-hive/README.md:
```diff
@@ -0,0 +1,17 @@
+# mapr52-hive 
+
+Docker image with HDFS, YARN and HIVE installed. Please note that running services have lower memory heap size set.
```
There's no HDFS in this image, only MapR FS.

petroav commented on teradatalabs/mapr52-hive/capabilities.txt:
```diff
@@ -0,0 +1,4 @@
+exposes_hive
```
What is this file for?

ArturGajowy replied:
So that the tests defined in https://github.com/Teradata/docker-images/blob/master/test/image_tests/image_tests.bats.sh are run for this image

petroav reviewed (commented):
Couple small comments.

ArturGajowy reviewed (changes requested):
The tests fail (see travis log). Seems like hive is not reachable on hadoop-master:10000. Please fix it and then we can proceed.

ArturGajowy commented:
Please make the PR only contain two commits, with the respective messages:
`Add mapr-base image`
`Add mapr-hive image`
both containing only the changes relevant accordting to their messages

ArturGajowy commented:
Please drop '52' from the image & directories names, we're not going to have two versions of the mapr container (say mapr-52 and mapr-53), so there's no point.

ArturGajowy commented on teradatalabs/mapr52-base/README.md:
```diff
@@ -0,0 +1,9 @@
+# mapr52-base
```
add the badges, as it is done for any other image. Just copy + paste then search & replace the image name

ArturGajowy commented on teradatalabs/mapr52-hive/README.md:
```diff
@@ -0,0 +1,17 @@
+# mapr52-hive 
```
ditto badges

ArturGajowy commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+# CONFIGURE MapR
+/opt/mapr/server/configure.sh  -N mycluster -Z localhost -C localhost -HS localhost -no-autostart
```
double space before -N

ArturGajowy commented on teradatalabs/mapr52-hive/files/wardenTracker.sh:
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+hname=$(hostname)
```
use CAPITAL_UNDERSCORE variable names, as is the de-facto standard for bash files

ArturGajowy replied:
don't use abbreviations for variable names. This should be HOSTNAME, not hname

ArturGajowy commented on teradatalabs/mapr52-hive/files/wardenTracker.sh:
```diff
@@ -0,0 +1,54 @@
+   maprcliReady=$(maprcli service list -node $hname | grep 'ERROR (10009)' | wc -l)
+   Services=0
+else
+   Services=$(maprcli service list -node $hname | grep CLDB |awk '{$1=$1};1' | tr ' ' '\n' | tail -1f)
```
use spaces around both sides of operators `|`, `grep CLDB |awk` in this case.

ArturGajowy replied:
extract a function for checking if the service started

ArturGajowy replied:
Do those services listen on ports? I think most of them do. Then use  `dockerize` for each of the services

ArturGajowy replied:
Once you extract the function for waiting for services, the reocurring 'WAIT FOR $FOO TO START' comments will not be needed anymore - remove them

ArturGajowy replied:
Actually, the 6 lines containing the dockerize calls waiting for the services should just go to `bootstrap.sh`, this file should be deleted

AS186170 replied:
dockerize will not work in this case , in MapR  ports open very late and there is no finite time to open port @ArturGajowy  , similar issue we faced in the test container also and i was forced to remove expose_hive and create new function expose_mapr

ArturGajowy commented on teradatalabs/mapr52-hive/files/wardenTracker.sh:
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
```
add -e

ArturGajowy replied:
use lowercase_underscore filenames for bash scripts

ArturGajowy commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,61 @@
+MAINTAINER Teradata Docker Team <docker@teradata.com>
+
+# ADD REPO FOR MapR
+ADD files/maprtech.repo /etc/yum.repos.d/maprtech.repo
```
Download the repo file on startup as is done in other containers. This way the source code contains information on where to get the `.repo` file, which can help future devs

ArturGajowy replied:
Do it in the dwonload in RUN below section

ArturGajowy commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,61 @@
+  && rpm --import http://package.mapr.com/releases/pub/maprgpg.key \
+
+# INSTALL UTILITY SOFTWARE
+  && yum install -y iputils vim openssh-server openssh-clients sudo \ 
```
all of this is not needed, remove (or prove that it's needed and not present in parent images anyway)

ArturGajowy commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,61 @@
+  && yum install -y iputils vim openssh-server openssh-clients sudo \ 
+
+# CONFIGURE SSH
+  && chkconfig sshd on \
```
move this to the `-hive` image, as is done in iop images (see iop-hive):
```


# SETUP SOCKS PROXY
RUN yum install -y openssh openssh-clients openssh-server
RUN ssh-keygen -t rsa -b 4096 -C "automation@teradata.com" -N "" -f /root/.ssh/id_rsa
RUN cp /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys
ADD files/socks-proxy.sh /root/socks-proxy.sh
```

This will ease pulling this up to a distro-agnostic base image (like centos6-...)

ArturGajowy commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,61 @@
+  && service sshd start \
+
+# INSTALL MapR 
+  && yum install -y mapr-fileserver mapr-nfs mapr-nodemanager  mapr-cldb \ 
```
do it in a single yum install command

ArturGajowy replied:
try to have just one yum install at all

ArturGajowy commented on teradatalabs/mapr52-base/Dockerfile:
```diff
@@ -0,0 +1,61 @@
+
+# CLEANUP
+  && yum -y clean all && rm -rf /tmp/* /var/tmp/* \
+  && ssh-keygen -t rsa -b 4096 -C "automation@teradata.com" -N "" -f /root/.ssh/id_rsa \
```
remove the following two lines. This should be in the 'setup socks proxy' section

ArturGajowy commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,52 @@
+MAINTAINER Teradata Docker Team <docker@teradata.com>
+
+# ADD ALL REQUIRED SCRIPTS AND FILES TO ROOT DIRECTORY
+ADD files/*.sh /root/
```
add each file verbatim, don't use *. It's better if we keep this explicit

ArturGajowy commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,52 @@
+ADD files/conf/hive-site.xml /opt/mapr/hive/hive-1.2/conf/hive-site.xml
+ADD files/conf/core-site.xml /opt/mapr/hadoop/hadoop-2.7.0/etc/hadoop/core-site.xml
+ADD files/supervisord.conf /etc/supervisord.conf
+COPY files/supervisord.d/* /etc/supervisord.d/
```
This should be done in the base image, see
```
# Copy supervisord startup script and base configs
COPY files/startup.sh /root/startup.sh
COPY files/supervisord.conf /etc/supervisord.conf
COPY files/supervisord.d/bootstrap.conf /etc/supervisord.d/bootstrap.conf

# Add supervisord configs in child images
ONBUILD COPY files/supervisord.d/* /etc/supervisord.d/
```
in any other base image

ArturGajowy commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,52 @@
+COPY files/supervisord.d/* /etc/supervisord.d/
+
+RUN chmod 777 /root/*.sh \
+# INSTALL UTILITY SOFTWARE  
```
don't install utility software, unless it's in a centos6-* image and we've all agreed it's needed

ArturGajowy commented on teradatalabs/mapr52-hive/Dockerfile:
```diff
@@ -0,0 +1,52 @@
+ADD files/supervisord.conf /etc/supervisord.conf
+COPY files/supervisord.d/* /etc/supervisord.d/
+
+RUN chmod 777 /root/*.sh \
```
why is that neeeded? Try to remove that or add a comment why it's needed

ArturGajowy commented on teradatalabs/mapr52-hive/files/bootstrap.sh:
```diff
@@ -0,0 +1,27 @@
+/opt/mapr/server/configure.sh  -N mycluster -Z localhost -C localhost -HS localhost -no-autostart
+
+# SETUP FLAT FILE /home/mapr/storagefile
+dd if=/dev/zero of=/home/mapr/storagefile bs=1G count=10
```
I'm not sure if this should be done here or in a setup.sh script. Probably depends on image size and the container startup time

ArturGajowy reviewed (changes requested):
First round done. Let me know when that's fixed and the tests pass on travis.

singhash committed:
Add Kerberize docker container for MapR with hive

Add Kerberize docker container for MapR with hive

singhash committed:
Add modification for docker-base

Add modification for docker-base

singhash committed:
Modification for kerberized MapR container

Modification for kerberized MapR container

singhash committed:
Adding modifications to test cluster for mapr

Adding modifications to test cluster for mapr

singhash committed:
Revert "Adding modifications to test cluster for mapr"

This reverts commit 586ebc734fac5509e64ad3b143e2807f15d985e7.

singhash committed:
Adding test container changes for mapr

Adding test container changes to accomodate mapr

singhash committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into mapr-52

singhash committed:
Adding ssh keys for mapr hadoop-master

Adding ssh keys for mapr hadoop-master

singhash committed:
Changes to accomodate tests for mapr

Changes to accomodate tests for mapr

singhash committed:
Adding Symlinks for the common files

Adding Symlinks for the common files

singhash committed:
Reverting links

Reverting links

petroav committed:
Add symlinks to mapr52-hive

petroav committed:
Fix links inside supervisord.d directory

singhash committed:
Modification to include env.sh changes

Modification to include env.sh changes in Kerberized MapR cluster

singhash committed:
Adding commands to start httpfs services

Adding commands to start httpfs services

singhash committed:
Adding privileged in docker-compose for MapR

Adding privileged in docker-compose for MapR

singhash committed:
Changing Test container to accomodate mapr

Changing Test container to accomodate mapr

singhash committed:
modification of tests to accomadate hive user

modification of tests to accomadate hive user

petroav committed:
Start sshd and the socks-proxy from bootstrap.sh

singhash committed:
Fixing kerberos container

Fixing kerberos container

singhash committed:
Changes for Debugging Bootsrap to know where its stuck

Debugging Bootsrap to know where its stuck

singhash committed:
Reverting changes for debugging bootstrap

Reverting changes for debugging bootstrap

singhash committed:
Moving partition creation to mapr-base

Moving partition creation to mapr-base and changes in capabilities

singhash committed:
decreasing allocation of space to 3 GB

decreasing allocation of space to 3 GB

singhash committed:
Increasing allocation of space back to 10 GB

Increasing allocation of space back to 10 GB

petroav committed:
Pick user to execute Hadoop commands based on image name

singhash committed:
makefile changes to debug images size

makefile changes to debug images size

singhash committed:
Changes in travis to increase wait time during build

Changes in travis to increase wait time during build

singhash committed:
Reverting the debugging changes

Reverting the debugging changes

singhash committed:
Adding httpfs to MapR-base

Adding httpfs to MapR-base and moving storagefile partition creation to
hive

singhash committed:
Removing creation of partition from kerberized container

Removing creation of partition from bootstrap of kerberized container
and removing chown

singhash committed:
Adding SSHD AND THE SOCKS PROXY for kerberized mapr

Adding SSHD AND THE SOCKS PROXY for kerberized mapr

singhash committed:
Rearranging the code in MapR cluster to make it ready for review

Rearranging the code in MapR cluster to make it ready for review

singhash committed:
Adding missing sshd service start in base

singhash committed:
Adding Logic to wait for MapR ticket in Warden

This changes will make maprcli wait for MapR ticket in wardenTracker

singhash committed:
changes in travis

changes in travis

AS186170 committed:
Adding Review Changes for MapR

singhash committed:
Travis yml changes

Travis yml changes

AS186170 commented:
@petroav @ArturGajowy  All the review comments are incorporated and all the tests are passing .
